doc/guides/porting/arch.rst

```diff
@@ -478,38 +568,189 @@ To enable the :option:`CONFIG_HW_STACK_PROTECTION` feature, the system must
 provide some kind of hardware-based stack overflow protection, and enable the
 :option:`CONFIG_ARCH_HAS_STACK_PROTECTION` option.
 
-There are no C APIs that need to be implemented to support stack protection,
-and it's entirely implemented within the ``arch/`` code.  However in most cases
-(such as if a guard region needs to be defined) the architecture will need to
-declare its own versions of the K_THREAD_STACK macros in ``arch/cpu.h``:
+Two forms of HW-based stack overflow detection are supported: dedicated
+CPU features for this purpose, or special read-only guard regions immediately
+preceding stack buffers.
 
-* :c:macro:`_ARCH_THREAD_STACK_DEFINE()`
-* :c:macro:`_ARCH_THREAD_STACK_ARRAY_DEFINE()`
-* :c:macro:`_ARCH_THREAD_STACK_MEMBER()`
-* :c:macro:`_ARCH_THREAD_STACK_SIZEOF()`
+:option:`CONFIG_HW_STACK_PROTECTION` only catches stack overflows for
+supervisor threads. This is not required to catch stack overflow from user
```

Comment:
Agree, but maybe clarify that user threads during system calls do require to catch stack overflows. Effectively, during sys calls they are sup threads, but maybe clarify this.

Reply:
Next paragraph now reads: "This feature only detects supervisor mode stack overflows, including stack
overflows when handling system calls"